source/pastemyst/web/paste.d

```diff
@@ -80,10 +80,11 @@ public class PasteWeb
         import pastemyst.db : tryFindOneById, createZip;
         import std.path : baseName;
 
-        // remove .zip form the id
-        string id = _id[0..($-".zip".length)];
+        // remove .zip form the id if present
+        if (_id.endsWith(".zip"))
+            _id = _id[0..($-".zip".length)];
```

Comment:
can you just enclose the if body in `{}` to stay consistent with the rest of the code?

Reply:
Should be there now.